test/unittest/xmlparser/XMLElementParserTests.cpp

```diff
@@ -4112,3 +4124,205 @@ TEST_F(XMLParserTests, env_var_substitution)
     clear_environment_variable(env_var_1);
     clear_environment_variable(env_var_2);
 }
+
+/*
+ * This test checks parsing of thread_settings elements.
+ */
+TEST_F(XMLParserTests, getXMLThreadSettings)
+{
+    /* Define the test cases */
+    std::vector<std::pair<std::vector<std::string>, XMLP_ret>> test_cases =
+    {
+        {{"12", "12", "12", "12", ""}, XMLP_ret::XML_OK},
+        {{"-1", "12", "12", "12", ""}, XMLP_ret::XML_OK},
+        {{"12", "-1", "12", "12", ""}, XMLP_ret::XML_OK},
+        {{"12", "12", "12", "-1", ""}, XMLP_ret::XML_OK},
+        {{"-2", "12", "12", "12", ""}, XMLP_ret::XML_ERROR},
+        {{"12", "12", "12", "-2", ""}, XMLP_ret::XML_ERROR},
```

Comment:
Checking for a negative affinity still pending:
```suggestion
        {{"-2", "12", "12", "12", ""}, XMLP_ret::XML_ERROR},
        {{"12", "12", "-2", "12", ""}, XMLP_ret::XML_ERROR},
        {{"12", "12", "12", "-2", ""}, XMLP_ret::XML_ERROR},
```

Reply:
a807237